qlog/qlog_dir.go

```diff
@@ -0,0 +1,58 @@
+package qlog
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/quic-go/quic-go/internal/utils"
+	"github.com/quic-go/quic-go/logging"
+)
+
+// QlogDir contains the value of the QLOGDIR environment variable.
+// If it is the empty string ("") no qlog output is written.
+var QlogDir string
+
+func init() {
+	QlogDir = os.Getenv("QLOGDIR")
```

Comment:
Should we read this outside of `init`? The last couple of times when I introduced environment variables, sooner or later people wanted to set them programmatically (`os.Setenv`), and that doesn't work if they're processed on initialization.

Reply:
This would make the code and the interface simpler. The performance impact should also be negligible.